sqlglot/parser.py

```diff
@@ -2738,6 +2738,9 @@ def _parse_column_def(self, this: t.Optional[exp.Expression]) -> t.Optional[exp.
 
         return self.expression(exp.ColumnDef, this=this, kind=kind, constraints=constraints)
 
+    def _parse_autoincrement(self) -> exp.Expression:
```

Comment:
this is not a parse, it's not matching anything, why is the snowflake one in it's own dialect?

Reply:
I factored this out so that it can be overridden easily, e.g. in Snowflake

Reply:
No problem with moving the snowflake method in the base parser

Reply:
how come?

Reply:
discussed in slack